database/migrations/2020_02_20_151112_add_connect_columns_to_users_table.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class AddConnectColumnsToUsersTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('roster', function (Blueprint $table) {            
+            // Add these to your users table
+            $table->text('access_token')->nullable();
+            $table->text('refresh_token')->nullable();
+            $table->unsignedBigInteger('token_expires')->nullable();
+        });
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table('roster', function (Blueprint $table) {
+            $table->dropColumn('token_expires');
+            $table->dropColumn('refresh_token');
+            $table->dropColumn('access_token');
```

Comment:
Again, reference my mention as to whether or not this is necessary for now.

I'm also not a fan of directly dropping the column in the down method because if the column wasn't able to be created it'll cause the migration to fail. The following is the best way I've been able to solve this problem and we have to do it for each column - It's kind of clunky, but Laravel doesn't give us a good way to do this.

```php
if (Schema::hasColumn('table_name', 'column_name')) {
    Schema::table('table_name', function (Blueprint $table) {
        $table->dropColumn('column_name');
    });
}
```

Reply:
We do still need these tokens - they're part of the session security. Good info on the possibility to fail migration - I'll add a check.